packages/components/src/local/asset-icon/index.tsx

```diff
@@ -10,13 +10,19 @@ import PropTypes from './types/props'
 import styles from './styles.module.scss'
 
 /* Render component */
-export const AssetIcon: React.FC<PropTypes> = ({ position, type, force, tooltip }: PropTypes) => {
+export const AssetIcon: React.FC<PropTypes> = ({ children, position, type, force, tooltip }) => {
+  const [showDialogue, setShowDialogue] = useState(false)
+
   const divIcon = L.divIcon({
     iconSize: [40, 40],
     className: cx(styles['asset-icon'], styles[force], styles[`platform-type-${type}`])
   })
-  return <Marker position={position} icon={divIcon} classNa>
+
+  const clickEvent = (): void => setShowDialogue(true)
```

Comment:
Aah, now I see how we do this declaratively.  I was just brainstorming what functionality we'll need, and thinking about where it could go: https://docs.google.com/drawings/d/1oLDcA2Vtf9awmcsKHjr6sGBhxZkY-4FrLaHuy1xnwDY/edit

The above line allows us to determine which dialog to show when an asset is clicked.